programs/main.cpp

```diff
@@ -238,8 +242,9 @@ void checkRequiredInstructionsImpl(volatile InstructionFail & fail)
 }
 
 /// This function is safe to use in static initializers.
-void writeError(const char * data, size_t size)
+void writeError(const char * data)
 {
+    size_t size = strlen(data);
```

Comment:
You must ensure that strlen is run at compile time (for example, this is true if you pass string literal), otherwise it will use SIMD instructions.

Reply:
Indeed, but before there was the code that does not guarantee this (strictly):

```c
        const char * msg2 = instructionFailToString(fail);
        writeError(msg2, strlen(msg2));
```

Ok, will take a look on how to guarantee this

Reply:
Done with a macro and assertion for __builtin_constant_p